### PR TITLE
feat: 添加斐波那契数列（e-maxx翻译

### PR DESCRIPTION
**Pull request by sshwy**
我把它的目录放在置换群的前面。因为这东西更好懂（？

### CODE REVIEW

File: docs/math/fibonacci.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+8 &=& 8 &=& F_6 &=& (000011)_F \\
+9 &=& 8 + 1 &=& F_6 + F_2 &=& (100011)_F \\
+19 &=& 13 + 5 + 1 &=& F_7 + F_5 + F_2 &=& (1001011)_F
+\end{eqnarray}
+$$
+
+给 $n$ 编码的过程可以使用贪心算法解决：
+
+1.  从大到小枚举斐波那契数 $F_i$ ，直到 $F_i\le n$ 。
+2.  把 $n$ 减掉 $F_i$ ，在编码的 $i-2$ 的位置上放一个 1。
```
Comment (Enter-tainer) [inline docs/math/fibonacci.md]: > Suppose this number was Fi. Subtract Fi from n and put a 1 in the i−2 position of the code word (indexing from 0 from the leftmost to the rightmost bit).

括号里面的东西还是挺有用的？

File: docs/math/fibonacci.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+3.  如果 $n$ 为正，回到步骤 1。
+4.  最后在编码末位添加一个 1，表示编码的结束位置。
+
+解码过程同理，先删掉末位的 1，对于编码为 1 的位置 $i$ ，累加一个 $F_{i+2}$ 到答案。最后的答案就是原数字。
```
Comment (Enter-tainer) [inline docs/math/fibonacci.md]: > To decode a code word, first remove the final 1. Then, if the i-th bit is set (indexing from 0 from the leftmost to the rightmost bit), sum Fi+2 to the number.

这里的括号同理

File: docs/math/fibonacci.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+## 斐波那契数列通项公式
+
+第 $n$ 个斐波那契数可以在 $\Theta (n)$ 的时间内使用递推公式计算。但我们仍有更快速的方法计算。
+
+### 解析解
+
+解析解即公式解。我们有斐波那契数列的通项公式：
```
Comment (Enter-tainer) [inline docs/math/fibonacci.md]: > There is a formula known as "Binet's formula", even though it was already known by Moivre:

这里的人名公式...emmm...需不需要加呢？

Comment (FFjet) [reply docs/math/fibonacci.md]: 这个就不需要了吧

Comment (Ir1d) [inline docs/math/fibonacci.md]:
```suggestion
解析解即公式解。我们有斐波那契数列的通项公式（Binet's Formula）：
```

**Comment by sshwy** [comment]
已经更新啦

**Pull request merged**